audplot/core/api.py

```diff
@@ -792,22 +792,49 @@ def waveform(
     default_figsize = plt.rcParams['figure.figsize']
     plt.rcParams['figure.figsize'] = (8, 1)
 
+    fig = plt.gcf()
     ax = ax or plt.gca()
 
     x = np.atleast_2d(x)
     channels, samples = x.shape
     if channels > 1:
         raise RuntimeError('Only mono signals are supported.')
```

Comment:
Not related to this PR, but when I saw this I was wondering why we are not automatically applying a mixdown and document this as behavior for multi-channel input.

Reply:
Yes and no. My take is that we should raise an error only if there is no reasonable way to recover. Otherwise maybe a warning would be sufficient, on the other hand, we almost never raise warnings and with a warning you never know if a user sees them :) Anyway, the error here is anyway not so critical as it very unlikely it will ever break some model training that runs for a week and then suddenly dies :)